velox/exec/Merge.cpp

```diff
@@ -165,60 +200,186 @@ RowVectorPtr Merge::getOutput() {
     return data;
   }
 
-  if (!output_) {
-    output_ = BaseVector::create<RowVector>(
-        outputType_, outputBatchSize_, operatorCtx_->pool());
-    for (auto& child : output_->children()) {
-      child->resize(outputBatchSize_);
+  VELOX_CHECK_NOT_NULL(sourceMerger_);
+  bool lastRun = false;
+  output_ = sourceMerger_->getOutput(
```

Comment:
How about we leverage output_ == nullptr to detect if have reached to the end of input?

Reply:
The `output_` might not be nullptr in the last run.
```C++
if (stream == nullptr) {
      if (outputSize > 0) {
        output_->resize(outputSize);
      }
      lastRun = true;
      return std::move(output_);
    }
```